### PR TITLE
TA 7599 move address widget to hui

### PR DESCRIPTION
**Pull request by bradparker**
### FilterSelect ###

Fuzzy filter a list of options and select with keyboard nav / mouse. You can supply custom Display and Option components.

screen shot 2015-10-07 at 11 22 29 am

screen shot 2015-10-07 at 11 23 21 am

screen shot 2015-10-07 at 11 23 31 am

screen shot 2015-10-07 at 11 24 05 am

### UrlSearchSelect ###

Provide a base URL, the param key for the TextInput value to go and some extra params and you can search any endpoint. Supports jsonp. Both the FilterSelect and the UrlSearchSelect use the same OptionList component for displaying / selecting supplied options.

screen shot 2015-10-07 at 11 25 30 am

screen shot 2015-10-07 at 11 25 48 am

screen shot 2015-10-07 at 11 26 02 am

###  CountrySelect ###

A FilterSelect preconfigured for choosing from a list of countries.

### AddressLookup ###

Built from a UrlSearchSelect (configured to search addresses) and a CountrySelect.

screen shot 2015-10-07 at 11 26 23 am

screen shot 2015-10-07 at 11 26 50 am

screen shot 2015-10-07 at 11 27 38 am

screen shot 2015-10-07 at 11 28 04 am

screen shot 2015-10-07 at 11 28 34 am

### AddressFieldset ###

screen shot 2015-10-07 at 11 28 45 am

### AddressFieldsetWithLookup ###

A combination of an AddressLookup and AddressFieldset

screen shot 2015-10-07 at 11 31 31 am

screen shot 2015-10-07 at 11 31 47 am

screen shot 2015-10-07 at 11 31 58 am

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,7 +46,9 @@
     "react": "^0.12.2",
     "react-router": "^0.11.5",
     "remarkable": "^1.6.0",
-    "superagent": "^1.2.0"
+    "superagent": "^1.2.0",
+    "superagent-jsonp": "0.0.5",
```
Comment (benoneal) [inline package.json]: Unused

File: lib/superagentJSONP.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  if (this._timeout && !this._timer) {
+    this._timer = setTimeout(() => {
+      this._jspCleanup()
+      this.timedout = true
+      this.abort()
+    }, this._timeout)
+  }
+}
+
+export default function superagentJSONP (request) {
```
Comment (benoneal) [inline lib/superagentJSONP.js]: Can we just use JSON?

**Comment by benoneal** [comment]
From our discussion Brad:
- [ ] Validate address breakdown fields when first shown, and on blur
- [ ] Show loading spinner on ID lookup
- [ ] Scroll list to keep selectionCandidate in view when using arrow keys 
- [ ] Scroll to selected item onOpen
- [ ] Set selectionCandidate to selected item onOpen
- [ ] Tab = close list and fill input with previous/current selection
- [ ] Enter/Space = close list and set new selection to selectionCandidate
- [ ] Enter/Tab behaviour also works when filter input focused (space should just type a space)
- [ ] Ensure it works in IE8/9

**Comment by bradparker** [comment]
@benoneal still couldn't quite get the DemoPage up and running in IE8, but will leave that for now and test in supporter.
Apart from that and scrolling with keyboard nav, everything else is there.

**Comment by bradparker** [comment]
@benoneal shoe-horned into Supporter for IE8 testing. Moved an event handler to another event and it's all good. This PR now includes everything needed to get this version into the US donation form too.

**Pull request merged**